operators/xr/README.md

```diff
@@ -0,0 +1,275 @@
+- **Memory Sharing**: Zero-copy memory between CUDA and Vulkan
+- **Command Buffering**: Pre-recorded Vulkan command buffers
+- **External Semaphores**: Cross-API synchronization primitives
+
+## Dependencies
+
+### External Libraries
+
+- **OpenXR SDK**: 1.0.26 - Cross-platform XR API
+- **OpenXR-Hpp**: 1.0.26 - C++ headers for OpenXR
+- **GLM**: 1.0.1 - Mathematics library for graphics
+- **Vulkan**: Graphics API for XR rendering
+- **CUDA**: Compute API for GPU operations
+
+### Holoscan Requirements
+
+- **Minimum SDK Version**: 3.1.0
+- **Tested Versions**: 3.1.0
+- **Platforms**: x86_64
+
+## Usage Example
+
+```python
+import holoscan as hs
+from holoscan.operators.xr import (
+    XrSession, XrBeginFrameOp, XrEndFrameOp,
+    XrCompositionLayerManager, XrSwapchainCuda
+)
+
+class XRApp(hs.Application):
+    def compose(self):
+        # Create XR session
+        xr_session = XrSession(
+            self,
+            application_name="My XR App",
+            application_version=1
+        )
+        
+        # Create composition layer manager
+        composition_manager = XrCompositionLayerManager(
+            self, xr_session=xr_session
+        )
+        
+        # Frame management operators
+        begin_frame = XrBeginFrameOp(
+            self, xr_session=xr_session
+        )
+        
+        end_frame = XrEndFrameOp(
+            self, xr_session=xr_session
+        )
+        
+        # Define workflow
+        self.add_flow(begin_frame, end_frame)
+```
+
+## Python API
+
+### Core Classes
+
+**XrSession**
+
+```python
+# Create XR session
+session = XrSession(
+    fragment,
+    application_name="My App",
+    application_version=1
+)
+
+# Access session properties
+view_configs = session.view_configurations()
+depth_range = session.view_configuration_depth_range()
+```
+
+**XrSwapchainCuda**
+
+```python
+# Create swapchain
+swapchain = XrSwapchainCuda(
+    session,
+    format=XrSwapchainCudaFormat.R8G8B8A8_SRGB,
+    width=1920, height=1080
+)
+
+# Acquire image for rendering
+tensor = swapchain.acquire()
+# ... render to tensor ...
+swapchain.release(cuda_stream)
+```
+
+**XrHandTracker**
+
+```python
+# Create hand tracker
+hand_tracker = XrHandTracker(
+    fragment,
+    xr_session=session,
+    hand=XrHandEXT.XR_HAND_LEFT_EXT
+)
+
+# Get hand joint locations
+joints = hand_tracker.locate_hand_joints()
+```
+
+### Data Types
+
+**Pose and Transform**
+
+```python
+# 3D position
+position = XrVector3f(x=0.0, y=0.0, z=0.0)
+
+# Quaternion rotation
+orientation = XrQuaternionf(x=0.0, y=0.0, z=0.0, w=1.0)
+
+# Complete pose
+pose = XrPosef(position=position, orientation=orientation)
+```
+
+**Frame State**
+
+```python
+# Frame timing information
+frame_state = XrFrameState(
+    predictedDisplayTime=1234567890,
+    predictedDisplayPeriod=16666667,
+    shouldRender=True
+)
+```
+
+## Building
+
+### Prerequisites
+
+- Holoscan SDK 3.1.0+
+- CUDA Toolkit
+- Vulkan SDK
+- CMake 3.20+
+
```

Comment:
Seems like these should be handled in the container build. Can we point to HoloHub top level README prerequisites instead?

```suggestion
```